contracts/oracles/ENSOracle.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./BaseChainlinkUSDOracle.sol";
+
+/// @notice Provides ENS/ETH price using ENS/USD and ETH/USD Chainlink oracles
+contract ENSOracle is BaseChainlinkUSDOracle {
+    constructor() BaseChainlinkUSDOracle(
+        0x5C00128d4d1c2F4f652C267d7bcdD7aC99C16E16, // ENS/USD
+        0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419, // ETH/USD
```

Comment:
Should ETH/USD oracle be hard-coded in the base-class constructor instead? It's called "USDOracle" so it doesn't seem like we intend it to be generic :)

Reply:
I changed it so now we have only one contract `chainlinkBasedOracle.sol` and depending on the construction arguments passed, we can create any other oracle that has underlying/USD chainlink feed. I left the ETH/USD to be passed as an argument so that the code can be reused on different networks (testnet including)